optimum/habana/transformers/training_args.py

```diff
@@ -648,12 +648,11 @@ def _setup_devices(self) -> "torch.device":
 
             if self.use_lazy_mode:
                 logger.info("Enabled lazy mode.")
-            # TODO: remove the block below when upgrade to SynapseAI 1.13 is done
-            # as eager mode will not be available anymore
             elif not self.torch_compile:
                 if os.getenv("PT_HPU_LAZY_MODE", "1") != "0":
-                    os.environ["PT_HPU_LAZY_MODE"] = "2"
-                logger.info("Enabled eager mode because use_lazy_mode=False.")
+                    assert (
+                        False
+                    ), "Lazy mode or compile mode not enabled => eager mode should be enabled using PT_HPU_LAZY_MODE=0"
```

Comment:
```suggestion
                    raise ValueError(
                        "Lazy mode or compile mode not enabled => eager mode should be enabled using PT_HPU_LAZY_MODE=0")
```
Let's rather raise an issue, I feel like it makes the error message clearer.